File: Python/BAITAP/Fibonacci.py
# Tinh day fibonacci: 0,1,1,2,3,5,8,13...
def Fibo(n):
    fn = int
    i = 3
    f1 = f2 = 1
    if n == 0:
        return 0
    if n == 1 or n == 2:
        return 1
    while i <= n:
        fn = f1 + f2
        f1 = f2
        f2 = fn
        i += 1
    return fn


n = int(input("Nhap N = "))
kq = Fibo(n)
print("F({0})={1}".format(n, kq))


# sd đệ quy
def Fib(x):
    if x <= 2:
        return 1
    else:
        return Fib(x - 1) + Fib(x - 2)


x = int(input("Nhap gia tri N:"))
result = Fib(x)
print("F({0})={1}".format(x, result))

File: C++/ThucHanh/BTRenLuyen/Fibonacci.cpp
#include<cmath>
#include<iostream>
using namespace std;
int fib(int n) {
	if(n<3) return 1;//diem dung
	return fib(n-1)+fib(n-2);//CT truy hoi tong quat sd de quy.Tinh day fibo tu fibo(1): 1,1,2,3,5,8,13...
}
int main() {
	int n;
	cout<<"Nhap n: ";
	cin>>n;
	int kq=fib(n);
	cout<<"Fibonacci("<<n<<")= "<<kq;
	return 0;
}

